optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-backend/src/main/java/org/optaplanner/workbench/screens/domaineditor/backend/server/validation/ScoreHolderUtils.java

```diff
@@ -0,0 +1,142 @@
+import org.optaplanner.core.api.score.buildin.hardmediumsoft.HardMediumSoftScore;
+import org.optaplanner.core.api.score.buildin.hardmediumsoft.HardMediumSoftScoreHolder;
+import org.optaplanner.core.api.score.buildin.hardmediumsoftbigdecimal.HardMediumSoftBigDecimalScore;
+import org.optaplanner.core.api.score.buildin.hardmediumsoftbigdecimal.HardMediumSoftBigDecimalScoreHolder;
+import org.optaplanner.core.api.score.buildin.hardmediumsoftlong.HardMediumSoftLongScore;
+import org.optaplanner.core.api.score.buildin.hardmediumsoftlong.HardMediumSoftLongScoreHolder;
+import org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore;
+import org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScoreHolder;
+import org.optaplanner.core.api.score.buildin.hardsoftbigdecimal.HardSoftBigDecimalScore;
+import org.optaplanner.core.api.score.buildin.hardsoftbigdecimal.HardSoftBigDecimalScoreHolder;
+import org.optaplanner.core.api.score.buildin.hardsoftdouble.HardSoftDoubleScore;
+import org.optaplanner.core.api.score.buildin.hardsoftdouble.HardSoftDoubleScoreHolder;
+import org.optaplanner.core.api.score.buildin.hardsoftlong.HardSoftLongScore;
+import org.optaplanner.core.api.score.buildin.hardsoftlong.HardSoftLongScoreHolder;
+import org.optaplanner.core.api.score.buildin.simple.SimpleScore;
+import org.optaplanner.core.api.score.buildin.simple.SimpleScoreHolder;
+import org.optaplanner.core.api.score.buildin.simplebigdecimal.SimpleBigDecimalScore;
+import org.optaplanner.core.api.score.buildin.simplebigdecimal.SimpleBigDecimalScoreHolder;
+import org.optaplanner.core.api.score.buildin.simpledouble.SimpleDoubleScore;
+import org.optaplanner.core.api.score.buildin.simpledouble.SimpleDoubleScoreHolder;
+import org.optaplanner.core.api.score.buildin.simplelong.SimpleLongScore;
+import org.optaplanner.core.api.score.buildin.simplelong.SimpleLongScoreHolder;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.io.IOService;
+
+@ApplicationScoped
+public class ScoreHolderUtils {
+
+    private IOService ioService;
+
+    private KieProjectService kieProjectService;
+
+    private ProjectClassLoaderHelper classLoaderHelper;
+
+    public ScoreHolderUtils() {
+    }
+
+    @Inject
+    public ScoreHolderUtils( @Named("ioStrategy") final IOService ioService,
+                             final KieProjectService kieProjectService,
+                             final ProjectClassLoaderHelper classLoaderHelper ) {
+        this.ioService = ioService;
+        this.kieProjectService = kieProjectService;
+        this.classLoaderHelper = classLoaderHelper;
+    }
+
+    public String extractScoreTypeFqn( final DataObject dataObject,
+                                       final Path dataObjectPath ) {
+        String superClassName = dataObject.getSuperClassName();
+
+        if ( superClassName.matches( "(\\w+\\.)*AbstractSolution<(\\w+\\.)*\\w+>" ) ) {
+            String scoreType = superClassName.substring( superClassName.indexOf( "<" ) + 1,
```

Comment:
You could use a capture group from your regex (if you wanted... and added it... but don't have to!)